src/libasr/pass/intrinsic_functions.h

```diff
@@ -2012,6 +2013,81 @@ namespace Fraction {
     }
 }  // namespace Fraction
 
+namespace SetExponent {
+    static ASR::expr_t *eval_SetExponent(Allocator &al, const Location &loc,
+            ASR::ttype_t* arg_type, Vec<ASR::expr_t*> &args, diag::Diagnostics& /*diag*/) {
+        ASR::ttype_t* arguement_type = expr_type(args[0]);
+        int32_t kind = extract_kind_from_ttype_t(arguement_type);
+        if (kind == 4) { 
+            float x = ASR::down_cast<ASR::RealConstant_t>(args[0])->m_r;
+            int32_t I = ASR::down_cast<ASR::IntegerConstant_t>(args[1])->m_n;
+            int32_t exponent;
+            if (x == 0.0) {
+                exponent = 0;
+                float result1 =  x * std::pow((2), (-1*(exponent)));
+                float result = result1 * std::pow((2), I);
+                return make_ConstantWithType(make_RealConstant_t, result, arg_type, loc);
+            }
+            else{
```

Comment:
Format things this way:
```suggestion
            } else {
```